src/modules/pages/HomePage.tsx

```diff
@@ -1,128 +1,48 @@
 import { useState } from 'react';
-import { useNavigate } from 'react-router';
 
-import { Divider, Skeleton, Stack, styled } from '@mui/material';
-
-import { Context, DiscriminatedItem } from '@graasp/sdk';
-import { Button, Main, MainMenu } from '@graasp/ui';
+import { Pagination, PaginationItem, Stack } from '@mui/material';
 
 import { usePlayerTranslation } from '@/config/i18n';
-import { buildMainPath } from '@/config/paths';
 import { hooks } from '@/config/queryClient';
 import {
-  HOME_NAVIGATION_STACK_ID,
-  MY_ITEMS_ID,
-  SHOW_MORE_ITEMS_ID,
+  HOME_PAGE_PAGINATION_ID,
+  buildHomePaginationId,
 } from '@/config/selectors';
 import { PLAYER } from '@/langs/constants';
 import PlayerCookiesBanner from '@/modules/cookies/PlayerCookiesBanner';
-import HeaderNavigation from '@/modules/header/HeaderNavigation';
-import HeaderRightContent from '@/modules/header/HeaderRightContent';
 import ItemGrid from '@/modules/main/ItemGrid';
-import TreeView from '@/modules/tree/TreeView';
-
-const StyledDivider = styled(Divider)(({ theme }) => ({
-  margin: theme.spacing(2, 0),
-}));
 
 const { useAccessibleItems } = hooks;
 
 const PAGE_SIZE = 20;
 
 const HomePage = (): JSX.Element => {
   const { t } = usePlayerTranslation();
-  const navigate = useNavigate();
 
   const [page, setPage] = useState(1);
```

Comment:
Is the page state still needed in the `Pagination` component, or it is possible to only pass the page as a props and to update the state in the parent using the callback ?

Reply:
I use a controlled version of the pagination component. So the state is needed here. It serves also to change the query key. I am not sure what you were suggesting to do however.